Source/Dafny/AST/DafnyAst.cs

```diff
@@ -13,14 +13,92 @@
 using System.Diagnostics.Contracts;
 using System.Numerics;
 using System.Linq;
-using Microsoft.Boogie;
 using System.Diagnostics;
 using System.Threading;
+using Microsoft.Boogie;
 
 namespace Microsoft.Dafny {
   [System.AttributeUsage(System.AttributeTargets.Field)]
   public class FilledInDuringResolutionAttribute : System.Attribute { }
 
+  public interface IToken : Microsoft.Boogie.IToken {
+    // Inherited for now.
+    /*
+    int kind { get; set; }
+    string filename { get; set; }
+    int pos { get; set; }
+    int col { get; set; }
+    int line { get; set; }
+    string val { get; set; }
+    bool IsValid { get; }*/
+    string leadingTrivia { get; set; }
+    string trailingTrivia { get; set; }
+  }
+
+  public record Token : IToken {
+    private int _kind;
+    private string _filename;
+    private int _pos;
+    private int _col;
+    private int _line;
+    private string _val;
+    private string _leadingTrivia;
+    private string _trailingTrivia;
+    public Token next;
```

Comment:
What is `next` ?

Reply:
> What is next ?

The token following the current token.

Reply:
I renamed it to `peekedToken` because it's an implementation detail in the scanning strategy. It's not exposed in the `IToken` interface.